lib/wasi/src/state/builder.rs

```diff
@@ -37,15 +38,18 @@ pub struct WasiStateBuilder {
     args: Vec<Vec<u8>>,
     envs: Vec<(Vec<u8>, Vec<u8>)>,
     preopens: Vec<PreopenedDir>,
+    vfs_preopens: Vec<String>,
     #[allow(clippy::type_complexity)]
     setup_fs_fn: Option<Box<dyn Fn(&mut WasiFs) -> Result<(), String> + Send>>,
-    stdout_override: Option<Box<dyn WasiFile>>,
-    stderr_override: Option<Box<dyn WasiFile>>,
-    stdin_override: Option<Box<dyn WasiFile>>,
+    stdout_override: Option<Box<dyn VirtualFile>>,
+    stderr_override: Option<Box<dyn VirtualFile>>,
+    stdin_override: Option<Box<dyn VirtualFile>>,
+    fs_override: Option<Box<dyn wasmer_vfs::FileSystem>>,
 }
 
 impl std::fmt::Debug for WasiStateBuilder {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // TODO: update this when stable
```

Comment:
⚠️ There is a todo here that must be addressed at the end of the review process.